### PR TITLE
Dev - SubmissionsForProcessing table to MongoDB collection 

### PR DESCRIPTION
**Pull request by jicata**
Rewire SubmissionsForProcessing table to make use of MongoDB and Hangfire https://github.com/SoftUni-Internal/suls-issues/issues/3016

### CODE REVIEW

File: Open Judge System/Data/OJS.Data/Repositories/ContestInstancesRepository.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public class ContestInstancesRepository : DeletableEntityRepository<ContestInstance>, IContestInstancesRepository
+    {
+        public ContestInstancesRepository(IOjsDbContext context)
+            : base(context)
+        {
+        }
+
+        public IQueryable<ContestInstance> AllActive()
+        {
+            return
```
Comment (rozay) [inline Open Judge System/Data/OJS.Data/Repositories/ContestInstancesRepository.cs]: Use `=>` in file

File: Open Judge System/Data/OJS.Data/Migrations/201709181417327_SubmissionsForProcessingDropTable.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                "dbo.SubmissionsForProcessing",
+                c => new
+                    {
+                        Id = c.Int(nullable: false, identity: true),
+                        SubmissionId = c.Int(nullable: false),
+                        Processing = c.Boolean(nullable: false),
+                        Processed = c.Boolean(nullable: false),
+                    })
+                .PrimaryKey(t => t.Id);
+
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/Migrations/201709181417327_SubmissionsForProcessingDropTable.cs]: no blank line here

File: Open Judge System/Data/OJS.Data/OjsData.cs [hunk 3]
```diff
@@ -49,7 +57,7 @@ public OjsData(IOjsDbContext context)
         public ISubmissionsRepository Submissions => (SubmissionsRepository)this.GetRepository<Submission>();
 
         public ISubmissionsForProcessingRepository SubmissionsForProcessing => (SubmissionsForProcessingRepository)
-            this.GetRepository<SubmissionForProcessing>();
+            this.GetMongoRepository<SubmissionForProcessing,ObjectId>();
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/OjsData.cs]: space after `,`

File: Open Judge System/Data/OJS.Data/OjsData.cs [hunk 4]
```diff
@@ -49,7 +57,7 @@ public OjsData(IOjsDbContext context)
         public ISubmissionsRepository Submissions => (SubmissionsRepository)this.GetRepository<Submission>();
 
         public ISubmissionsForProcessingRepository SubmissionsForProcessing => (SubmissionsForProcessingRepository)
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/OjsData.cs]: (SubmissionsForProcessingRepository) on next row pls

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ParticipantsController.cs [hunk 5]
```diff
@@ -256,5 +260,17 @@ public ActionResult ResetSubmissions()
 
             return null;
         }
+
+        public ActionResult DeleteProcessedSubmissions()
+        {
+            string cron = "0 3 * * *";
```
Comment (YanaSlavcheva) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ParticipantsController.cs]: var

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ParticipantsController.cs [hunk 6]
```diff
@@ -256,5 +260,17 @@ public ActionResult ResetSubmissions()
...
+
+            this.backgroundJobService.AddOrUpdateRecurringJob(
```
Comment (YanaSlavcheva) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ParticipantsController.cs]: How many times do you need to create this background job?

File: Open Judge System/Workers/OJS.Workers.LocalWorker/LocalWorkerService.cs [hunk 7]
```diff
@@ -97,9 +97,8 @@ private void ResetAllProcessingSubmissions()
                         foreach (var unprocessedSubmission in allProcessingSubmissions)
                         {
                             unprocessedSubmission.Processing = false;
+                            data.SubmissionsForProcessing.Update(unprocessedSubmission);
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.LocalWorker/LocalWorkerService.cs]: this?

File: Open Judge System/Workers/OJS.Workers.LocalWorker/SubmissionJob.cs [hunk 8]
```diff
@@ -81,7 +81,7 @@ public void Start()
                             if (submission != null && submissionForProcessing != null && !submission.Processing)
                             {
                                 submissionForProcessing.Processing = true;
-                                data.SaveChanges();
+                                data.SubmissionsForProcessing.Update(submissionForProcessing);
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.LocalWorker/SubmissionJob.cs]: this?

File: Open Judge System/Workers/OJS.Workers.LocalWorker/SubmissionJob.cs [hunk 9]
```diff
@@ -122,7 +122,9 @@ public void Start()
                 }
 
                 submission.Processed = true;
-                data.SubmissionsForProcessing.Delete(submissionForProcessing);
+                submissionForProcessing.Processed = true;
+                submissionForProcessing.Processing = false;
+                data.SubmissionsForProcessing.Update(submissionForProcessing);
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.LocalWorker/SubmissionJob.cs]: this?

**Comment by YanaSlavcheva** [comment]
We may be free to delete this PR and branch since we will not implement Mongo? @jicata ?

**Comment by jicata** [comment]
Absolutely so.

**Pull request closed without merging**